packages/core/lib/migrate.ts

```diff
@@ -89,6 +107,55 @@ const migrations: Migration[] = [
       delete updated.data.zones?.[zoneCompound];
     });
 
+    // Migrate zones created by dynamic arrays
+    const unmigratedZonesGrouped: Record<string, Record<string, Content>> = {};
+
+    Object.keys(updated.data.zones ?? {}).forEach((zoneCompound) => {
+      const [zoneId, propName] = zoneCompound.split(":");
```

Comment:
First part is component id, not zone id

```suggestion
      const [componentId, propName] = zoneCompound.split(":");
```